openpype/pipeline/file_collection.py

```diff
@@ -0,0 +1,274 @@
+"""Utility functions for file sequences and collections
+"""
+import os
+import re
+import clique
+from openpype.lib import Logger
+from openpype.pipeline import frames
+
+
+def collect_filepaths_from_sequential_path(
+    file_path,
+    frame_start=None,
+    frame_end=None,
+    only_existing=False,
+):
+    """Generate expected files from path
+
+    Args:
+        file_path (str): Path to generate expected files from
+        frame_start (Optional[int]): Start frame of the sequence
+        frame_end (Optional[int]): End frame of the sequence
+        only_existing (Optional[bool]): Ensure that files exists.
+
+    Returns:
+        Any[list[str], str]: List of absolute paths to files
+            (frames of a sequence) or path if single file
+    """
+
+    dirpath = os.path.dirname(file_path)
+    filename = os.path.basename(file_path)
+
+    formattable_string = convert_filename_to_formattable_string(
+        filename)
+
+    if (
+        not formattable_string
+        or frame_start is None
+        or frame_end is None
+    ):
+        return file_path
+
+    expected_files = []
+    for frame in range(int(frame_start), (int(frame_end) + 1)):
+        frame_file_path = os.path.join(
+            dirpath, formattable_string.format(frame)
+        )
+        # normalize path
+        frame_file_path = os.path.normpath(frame_file_path)
+
+        # make sure file exists if ensure_exists is enabled
+        if only_existing and not os.path.exists(frame_file_path):
+            continue
+
+        # add to expected files
+        expected_files.append(
+            # add normalized path
+            frame_file_path
+        )
+
+    return expected_files
+
+
+def generate_expected_filepaths(
+    frame_start,
+    frame_end,
+    path,
+):
+    """Generate expected files from path
+
+    Args:
+        frame_start (int): Start frame of the sequence
+        frame_end (int): End frame of the sequence
+        path (str): Path to generate expected files from
+
+    Returns:
+        Any[list[str], str]: List of expected absolute paths to files
+            (frames of a sequence) or path if single file
+    """
+    return collect_filepaths_from_sequential_path(
+        frame_start,
+        frame_end,
+        path,
+        only_existing=False,
+    )
```

Comment:
It's funny that this function calls `collect_filepaths_from_sequential_path` even though I'd have expected it to be the opposite:
```python
def collect_filepaths_from_sequential_path(*args, **kwargs):
    expected = generate_expected_files(*args, **kwargs)
    return [path for path in expected if os.path.exists(path)]
```

To me it makes no sense by the way that this will return either a single filepath or a list of files. I'd expect it to always return a list of files.